server/src/middleware/recipeSchema.ts

```diff
@@ -0,0 +1,23 @@
+import Joi from "joi";
+
+export const recipeSchemas = {
+  recipePOST: Joi.object().keys({
+    title: Joi.string().required(),
+    description: Joi.string().required(),
+    categories: Joi.array().items(Joi.string()).required(),
+    ingredients: Joi.array().items(Joi.string()).required(),
+    steps: Joi.array().items(Joi.string()).required(),
+  }),
+  recipePATCH: Joi.object().keys({
+    title: Joi.string().optional(),
+    description: Joi.string().optional(),
+    categories: Joi.array().items(Joi.string()).optional(),
+    ingredients: Joi.array().items(Joi.string()).optional(),
+    steps: Joi.array().items(Joi.string()).optional(),
+  }),
+  recipeDetails: {
+    _id: Joi.string()
```

Comment:
while this can work, I suggest to consider creating a custom validation function like described here: https://joi.dev/api/?v=17.9.1#anycustommethod-description

In that way you can use `ObjectId` class from mongodb driver like you did before.